README.md

```diff
@@ -1,6 +1,7 @@
-# WireGuard Tools
+# WireGuard Key Exchange
 
-WireGuard Tools consists of both a frontend and a backend to dynamically supply WireGuard keys to a FreiFunk Gateway offering BATMAN over VXLAN over WireGuard.
+WireGuard Key Exchange is a tool consisting of two parts: a frontend (broker) and a backend (worker). The frontend (broker) is where the client can push (register) its key before connecting. The backend (worker) is injecting those keys into a WireGuard instance.
+This tool is intended to facilitate running BATMAN over VXLAN over WireGuard as a means to create encrypted high-performance mesh links.
```

Comment:
That should probably be opt-in, to keep the tooling generic.

Reply:
I think that BATMAN o VXLAN o WG is the major usecase, and the initial reason to start this project, so it deserves a spot in the README as to why we are doing this.
In practice the exact behaviour (only inject keys into a matching wg instance, also do some VXLAN related work, etc) could be made configurable?
Unless someone has a concrete replacement for this line I'd let it sit there until we have more actual code.

Reply:
wfm